lib/oli/delivery/depot.ex

```diff
@@ -113,4 +113,29 @@ defmodule Oli.Delivery.Depot do
     |> :ets.select([match_spec])
     |> Serializer.unserialize(depot_desc)
   end
+
+  @doc """
+  Returns the count of entries that match the given conditions.
+
+  Full details on the supported conditions can be found in the MatchSpecTranslator
+  module, but the basic idea is that you can pass a list of conditions to match on.
+
+  The list can be a simple keyword list of field values to match on as an AND:
+
+  [resource_type_id: 3, graded: false]
+
+  or a list of tuples with the field name, operator, and value (all of which are
+  ANDed together):
+
+  [{:duration, {:between, 5, 10}}, {:graded, {:=, true}}]
+  """
+  def count(%DepotDesc{} = depot_desc, table_id, conditions) do
```

Comment:
This seems like a nice addition to the core `Depot`, but I don't see it being used anywhere in this PR.  Also, if we want to keep it - we should add some unit tests for it.

Reply:
Yes, you are right. At first, I built this new depot function to use it [here
](https://github.com/Simon-Initiative/oli-torus/pull/5225/commits/8358f26903120a72ce377ffe2cfd6c028529a9be#diff-9cd70d1d862781e5ecd2a468b858c410ad804e68e0bb40ea8d86491a192649d9R106-R116)
But then I took another approach and deleted it in [this commit](https://github.com/Simon-Initiative/oli-torus/pull/5225/commits/55210bebb6046b92f862af35389360a416e741c8), but decided to keep the Depot function because I think it will probably be useful in the future. So, if you agree, I can add some tests for it and keep it, WDYT?

Reply:
Yes, let's keep it